src/DitStr.jl

```diff
@@ -159,6 +159,117 @@ Base.@propagate_inbounds function Base.getindex(dit::DitStr{D,N,T}, itr::Abstrac
     return map(x -> readat(dit, x), itr)
```

Comment:
> This feature seems to be useful, and is throughly tested. I am wondering if it is beneficial to add some eye candies like `bit_string[start:stop]`?

Sounds great but we already have a slicing method which returns `Vector`-type results. So I'm hestitating whether to change the output of this function.

Reply:
Additionally, this feature is different from `String` slicing, since we don't return a copied slice. This is more efficient for `BitBasis` as `bit_strings` are immutable. 
In Julia's Base Library, they use macro `@views` to automatically expand `s[i:j]` to `SubString` type as inplementated in this PR.

Reply:
You are right, Julia slicing returns a copy. Then I agree we do not need the slicing operation.